consensus/core/src/block.rs

```diff
@@ -39,6 +40,31 @@ impl fastcrypto::hash::Hash<{ DIGEST_LENGTH }> for Block {
     }
 }
 
+impl PartialEq for Block {
+    fn eq(&self, other: &Self) -> bool {
+        self.reference() == other.reference()
+    }
+}
+
+impl Eq for Block {}
+
+impl fmt::Debug for Block {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self)
+    }
+}
+
+impl fmt::Display for Block {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}:[", self.reference())?;
+        for ancestor in self.ancestors() {
```

Comment:
I think printing ancestors and transactions should only be in Debug, not in Display.